sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -168,32 +189,45 @@ def run_query(query, args, pipeline_options):
       result.cancel()
     else:
       result.wait_until_finish()
-  except NotImplementedError as exp:
-    print(exp)
+  except Exception as exc:
+    thread_errors.append(str(exc))
```

Comment:
What happens if `thread_errors` is None. This will cause an exception itself. Also after appending a better practice would be to raise here.

Another minor comment is, you could append the whole excpetion instead of its string representation.

Reply:
Added the `raise`. Re-raising in a thread doesn't pass the exception to the main thread, but it's useful for the case of DataflowRunner since run_query doesn't run in a thread.